File: bildunya_frontend/lib/data/models/conversation_dto.dart
/// [com.bildunya.dto.ConversationDto]
class ConversationDto {
  const ConversationDto({
    this.id,
    this.user1Id,
    this.user2Id,
    this.otherUserId,
    this.otherUsername,
    this.otherFullName,
    this.createdAt,
  });

  final int? id;
  final int? user1Id;
  final int? user2Id;
  final int? otherUserId;
  final String? otherUsername;
  final String? otherFullName;
  final String? createdAt;

  factory ConversationDto.fromJson(Map<String, dynamic> json) {
    return ConversationDto(
      id: (json['id'] as num?)?.toInt(),
      user1Id: (json['user1_id'] as num?)?.toInt() ??
          (json['user1Id'] as num?)?.toInt(),
      user2Id: (json['user2_id'] as num?)?.toInt() ??
          (json['user2Id'] as num?)?.toInt(),
      otherUserId: (json['other_user_id'] as num?)?.toInt() ??
          (json['otherUserId'] as num?)?.toInt(),
      otherUsername: json['other_username'] as String? ??
          json['otherUsername'] as String?,
      otherFullName: json['other_full_name'] as String? ??
          json['otherFullName'] as String?,
      createdAt: json['created_at'] as String? ?? json['createdAt'] as String?,
    );
  }
}

File: BilDunya/src/main/java/com/bildunya/dto/ConversationDto.java
package com.bildunya.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ConversationDto {

    private Long id;

    @JsonProperty("user1_id")
    private Long user1Id;

    @JsonProperty("user2_id")
    private Long user2Id;

    @JsonProperty("other_user_id")
    private Long otherUserId;

    @JsonProperty("other_username")
    private String otherUsername;

    @JsonProperty("other_full_name")
    private String otherFullName;

    @JsonProperty("created_at")
    private String createdAt;

    @JsonProperty("related_content_id")
    private Long relatedContentId;

    @JsonProperty("related_content_label")
    private String relatedContentLabel;
}

